examples/example-auto-router/src/app/prepare.ts

```diff
@@ -33,9 +37,19 @@ const prepare = async () => {
 		subgraphIndexRepo(await account.chainId())
 	);
 
+	const savingsVault = new SavingsVault(
+		account,
+		getEnv('SAVINGS_VAULT_ADDRESS')
+	);
+
 	const autoRouter = new AutoRouter(indexRouter, zeroEx);
 
-	return { account, index, autoRouter };
+	const metaRouter = new MetaRouter(new SavingsVaultRouter(), autoRouter, {
+		[ProductType.INDEX]: [index.address],
+		[ProductType.SAVINGS_VAULT]: [savingsVault.address],
+	});
```

Comment:
It is more optimized, but not relevant for this use case, there will be a couple of routers and a couple of addresses, so I don't see any benefit in refactoring this.

Reply:
agree with @jn-lp, let's move on with more optimized approach

Reply:
```
const metaRouter = new MetaRouter(new SavingsVaultRouter(), autoRouter, {
	  [index.address]: ProductType.INDEX,
	  [savingsVault.address]: ProductType.SAVINGS_VAULT,
	});
```
I don't see that this reduces the complexity since to find which product type (Index or SavingsVault) the address belongs to, I have to loop over both of the arrays anyways. 

```
const metaRouter = new MetaRouter({
	  [index.address]: autoRouter,
	  [savingsVault.address]: new SavingsVaultRouter(),
	 });
```

And it is the same in this case. 

Am I not seeing something, @jn-lp @HvrlK?

Reply:
Ok, typescript weird syntax, [] brackets indicate the type. I thought it is something else, thus the confusion. I agree it makes sense to use a map here for 0(1) access.